task/uartecho/src/main.rs

```diff
@@ -190,8 +190,11 @@ fn configure_uart_device() -> Usart {
 
     let hardware_flow_control = cfg!(feature = "hardware_flow_control");
 
+    #[allow(clippy::needless_late_init)]
     let usart;
+    #[allow(clippy::needless_late_init)]
     let peripheral;
+    #[allow(clippy::needless_late_init)]
```

Comment:
This would be pretty easy to actually fix – delete these lines and add `let`s in each `cfg_if` branch below.

Reply:
I also like the extra `let`s if others agree with the style